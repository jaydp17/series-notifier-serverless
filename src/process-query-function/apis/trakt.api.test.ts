/**
 * Tests the trakt api
 */

import * as TraktAPI from './trakt.api';

describe('TraktAPI', () => {
  it('search flash', async () => {
    const result = await TraktAPI.searchShow('flash');
    expect(result).toMatchSnapshot();
  });
});
