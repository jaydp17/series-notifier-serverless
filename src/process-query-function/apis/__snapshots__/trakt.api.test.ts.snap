// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TraktAPI search flash 1`] = `
Array [
  Object {
    "score": 1038.9548,
    "show": Object {
      "aired_episodes": 64,
      "airs": Object {
        "day": "Tuesday",
        "time": "20:00",
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "ar",
        "bg",
        "bs",
        "cs",
        "da",
        "de",
        "el",
        "en",
        "es",
        "fa",
        "fr",
        "he",
        "hr",
        "hu",
        "id",
        "it",
        "ko",
        "lb",
        "lt",
        "ml",
        "ms",
        "nl",
        "pl",
        "pt",
        "ro",
        "ru",
        "sk",
        "sr",
        "sv",
        "te",
        "th",
        "tr",
        "uk",
        "vi",
        "za",
        "zh",
      ],
      "certification": "TV-PG",
      "country": "us",
      "first_aired": "2014-10-07T00:00:00.000Z",
      "genres": Array [
        "drama",
        "fantasy",
        "science-fiction",
        "action",
        "adventure",
      ],
      "homepage": "http://www.cwtv.com/shows/the-flash/",
      "ids": Object {
        "imdb": "tt3107288",
        "slug": "the-flash-2014",
        "tmdb": 60735,
        "trakt": 60300,
        "tvdb": 279121,
        "tvrage": 36939,
      },
      "language": "en",
      "network": "The CW",
      "overview": "Barry Allen wakes up nine months after he was struck by lightning and discovers that the bolt gave him the power of super speed. With his new team and powers, Barry becomes \\"The Flash\\" and fights crime in Central City.",
      "rating": 8.29135,
      "runtime": 45,
      "status": "returning series",
      "title": "The Flash",
      "trailer": "http://youtube.com/watch?v=aZxhA_Jihmw",
      "updated_at": "2017-04-23T10:41:26.000Z",
      "votes": 13379,
      "year": 2014,
    },
    "type": "show",
  },
  Object {
    "score": 23.684973,
    "show": Object {
      "aired_episodes": 22,
      "airs": Object {
        "day": "Tuesday",
        "time": "20:00",
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "bg",
        "bs",
        "da",
        "de",
        "en",
        "fr",
        "hu",
        "it",
        "pt",
        "ro",
        "ru",
      ],
      "certification": "TV-PG",
      "country": "us",
      "first_aired": "1990-09-20T00:00:00.000Z",
      "genres": Array [
        "drama",
        "action",
        "crime",
        "fantasy",
        "adventure",
      ],
      "homepage": "http://www.dccomics.com/tv/the-flash-1990-1991",
      "ids": Object {
        "imdb": "tt0098798",
        "slug": "the-flash",
        "tmdb": 236,
        "trakt": 235,
        "tvdb": 78650,
        "tvrage": 5781,
      },
      "language": "en",
      "network": "CBS",
      "overview": "Central City Police forensic scientist Barry Allen's crime lab is struck by lightning. Allen's electrified body is flung into and shatters a cabinet of chemicals, which are both electrified and forced to interact with each other and with his physiology when they come into physical contact with his body. He soon discovers that the accident has changed his body's metabolism and as a result he has gained the ability to move at superhuman speed. Barry Allen has become the Flash.",
      "rating": 7.26885,
      "runtime": 60,
      "status": "ended",
      "title": "The Flash",
      "trailer": null,
      "updated_at": "2017-04-22T09:59:29.000Z",
      "votes": 305,
      "year": 1990,
    },
    "type": "show",
  },
  Object {
    "score": 8.775088,
    "show": Object {
      "aired_episodes": 22,
      "airs": Object {
        "day": "Friday",
        "time": "21:00",
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "bs",
        "en",
        "es",
        "ja",
      ],
      "certification": "TV-PG",
      "country": "us",
      "first_aired": "2007-08-10T07:00:00.000Z",
      "genres": Array [
        "comedy",
        "drama",
        "action",
        "adventure",
        "fantasy",
        "science-fiction",
      ],
      "homepage": "http://www.scifi.com/flashgordon/",
      "ids": Object {
        "imdb": "tt0959086",
        "slug": "flash-gordon",
        "tmdb": 1225,
        "trakt": 1219,
        "tvdb": 80441,
        "tvrage": null,
      },
      "language": "en",
      "network": "SciFi",
      "overview": "Placing a 21st century spin on a science fiction classic, SCI FI's contemporized version of Flash Gordon follows the all-new adventures of Flash and his companions, Dale Arden, Baylin and Dr. Hans Zarkov. Ordinary people thrust into extraordinary circumstances, they find themselves as Earth's last line of defense against the forces of the merciless dictator Ming.",
      "rating": 6.15044,
      "runtime": 45,
      "status": "ended",
      "title": "Flash Gordon",
      "trailer": null,
      "updated_at": "2017-02-22T10:39:30.000Z",
      "votes": 113,
      "year": 2007,
    },
    "type": "show",
  },
  Object {
    "score": 5.4358954,
    "show": Object {
      "aired_episodes": 1,
      "airs": Object {
        "day": "Monday",
        "time": "21:10",
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "de",
        "en",
      ],
      "certification": "TV-14",
      "country": "us",
      "first_aired": "1996-09-08T01:10:00.000Z",
      "genres": Array [
        "comedy",
      ],
      "homepage": null,
      "ids": Object {
        "imdb": "tt0115173",
        "slug": "flash-forward",
        "tmdb": 10371,
        "trakt": 10320,
        "tvdb": 77954,
        "tvrage": 3556,
      },
      "language": "en",
      "network": "Disney Channel",
      "overview": "Becca Fisher and Tucker James are two 13 year-olds who have been best friends since birth. They live next door to each other and have shared birthday parties, babysitters and backyard swings. But now, Becca and Tucker are beginning the rite of passage known as the eighth grade. Suddenly, they are teenagers, testing the bounds of their new independence while struggling to hang on the their past.",
      "rating": 7.28571,
      "runtime": 22,
      "status": "returning series",
      "title": "Flash Forward",
      "trailer": null,
      "updated_at": "2017-02-24T15:12:27.000Z",
      "votes": 70,
      "year": 1996,
    },
    "type": "show",
  },
  Object {
    "score": 1.6061245,
    "show": Object {
      "aired_episodes": 29,
      "airs": Object {
        "day": "Saturday",
        "time": "22:00",
        "timezone": "Asia/Tokyo",
      },
      "available_translations": Array [
        "ar",
        "cs",
        "da",
        "de",
        "el",
        "en",
        "es",
        "fa",
        "fr",
        "he",
        "it",
        "ja",
        "ko",
        "pt",
        "ru",
        "sv",
        "uk",
        "zh",
      ],
      "certification": "TV-MA",
      "country": "jp",
      "first_aired": "2013-04-06T00:00:00.000Z",
      "genres": Array [
        "drama",
        "horror",
        "action",
        "adventure",
        "fantasy",
        "anime",
        "animation",
      ],
      "homepage": "http://shingeki.tv/",
      "ids": Object {
        "imdb": "tt2560140",
        "slug": "attack-on-titan",
        "tmdb": 1429,
        "trakt": 1420,
        "tvdb": 267440,
        "tvrage": null,
      },
      "language": "ja",
      "network": "MBS",
      "overview": "Several hundred years ago, humans were nearly exterminated by the Titans. Titans are typically several stories tall, seem to have no intelligence, devour human beings, and worst of all, seem to do it for the pleasure rather than as a food source. A small percentage of humanity survived by building a city protected by extremely high walls, even taller than the largest of Titans. Flash forward to the present and the city has not seen a Titan in over a hundred years, until one day, a Colossal Titan appears out of thin air and destroys part of the city wall. As teenage boy, Eren Jaeger, and his foster sister, Mikasa Ackerman, witness the destruction of their town and death of their mother at the hands of the Titans, Eren vows to kill every single Titan and take revenge for all of mankind.",
      "rating": 8.7668,
      "runtime": 25,
      "status": "returning series",
      "title": "Attack on Titan",
      "trailer": "http://youtube.com/watch?v=luYOt2-c2TI",
      "updated_at": "2017-04-23T11:50:04.000Z",
      "votes": 3928,
      "year": 2013,
    },
    "type": "show",
  },
  Object {
    "score": 0.6079643,
    "show": Object {
      "aired_episodes": 227,
      "airs": Object {
        "day": "Thursday",
        "time": null,
        "timezone": "Asia/Tokyo",
      },
      "available_translations": Array [
        "de",
        "el",
        "en",
        "es",
        "fr",
        "pl",
        "pt",
        "zh",
      ],
      "certification": null,
      "country": "jp",
      "first_aired": "1983-10-13T07:00:00.000Z",
      "genres": Array [
        "anime",
      ],
      "homepage": "http://www.tv-tokyo.co.jp/anime/cap_tsuba/",
      "ids": Object {
        "imdb": "tt0294023",
        "slug": "captain-tsubasa",
        "tmdb": 25707,
        "trakt": 25592,
        "tvdb": 84045,
        "tvrage": null,
      },
      "language": "ja",
      "network": "TV Tokyo",
      "overview": "Tsubasa Ozora moves to the town Nankatsu to increase his skills as a soccer player. He is challanged by the Super Goalkeeper Genzo Wakabayashi. He then agree to compete and see who was the best soccer player there.",
      "rating": 7.61481,
      "runtime": 25,
      "status": "ended",
      "title": "Captain Tsubasa",
      "trailer": null,
      "updated_at": "2017-04-16T10:48:22.000Z",
      "votes": 135,
      "year": 1983,
    },
    "type": "show",
  },
  Object {
    "score": 0.32615376,
    "show": Object {
      "aired_episodes": 32,
      "airs": Object {
        "day": null,
        "time": null,
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "de",
        "en",
        "pt",
      ],
      "certification": "TV-PG",
      "country": "us",
      "first_aired": "1979-09-08T07:00:00.000Z",
      "genres": Array [
        "animation",
      ],
      "homepage": "http://www.hearstent.com/flashgordon.html",
      "ids": Object {
        "imdb": "tt0078662",
        "slug": "the-new-adventures-of-flash-gordon",
        "tmdb": 11844,
        "trakt": 11791,
        "tvdb": 82452,
        "tvrage": null,
      },
      "language": "en",
      "network": "NBC",
      "overview": "Flash Gordon blasts off to the planet Mongo with girlfriend Dale Arden and scientist Hans Zarkov to prevent evil dictator Ming the Merciless from dominating the universe. In attempting to put an end to Ming's villainy, Flash receives the aid (and often the hindrance) of Prince Barin of Arboria, the Hawk Men led by King Vultan, Queen Fria of the ice-covered land of Frigia, Thun the Lion Man and Ming's sexy, scantily-clad daughter Aura, who has a powerful attraction for Flash.",
      "rating": 9.14286,
      "runtime": 30,
      "status": "ended",
      "title": "The New Adventures of Flash Gordon",
      "trailer": null,
      "updated_at": "2016-11-30T10:50:31.000Z",
      "votes": 7,
      "year": 1979,
    },
    "type": "show",
  },
  Object {
    "score": 0.3106226,
    "show": Object {
      "aired_episodes": 40,
      "airs": Object {
        "day": null,
        "time": null,
        "timezone": "Europe/London",
      },
      "available_translations": Array [],
      "certification": null,
      "country": null,
      "first_aired": "1936-04-06T08:00:00.000Z",
      "genres": Array [],
      "homepage": null,
      "ids": Object {
        "imdb": null,
        "slug": "flash-gordon-1936-65589",
        "tmdb": null,
        "trakt": 65589,
        "tvdb": 254867,
        "tvrage": null,
      },
      "language": null,
      "network": null,
      "overview": "Flash Gordon, Flash Gordon's Trip to Mars and Flash Gordon Conquers the Universe, the most expensive and popular movie serials ever made, have been favorites of movie and comic fans for decades.",
      "rating": 6.75,
      "runtime": 20,
      "status": "ended",
      "title": "Flash Gordon",
      "trailer": null,
      "updated_at": "2017-01-20T10:44:36.000Z",
      "votes": 4,
      "year": 1936,
    },
    "type": "show",
  },
  Object {
    "score": 0.3106226,
    "show": Object {
      "aired_episodes": 16,
      "airs": Object {
        "day": "Friday",
        "time": "21:00",
        "timezone": "Asia/Tokyo",
      },
      "available_translations": Array [],
      "certification": null,
      "country": "jp",
      "first_aired": "1994-01-21T00:00:00.000Z",
      "genres": Array [
        "animation",
      ],
      "homepage": null,
      "ids": Object {
        "imdb": "tt0109633",
        "slug": "dirty-pair-flash",
        "tmdb": null,
        "trakt": 64804,
        "tvdb": 73312,
        "tvrage": null,
      },
      "language": null,
      "network": "Showtime",
      "overview": "Dirty Pair Flash is related to the original Dirty Pair, with the same characters but there are some changes. Dirty Pair Flash was released later but it is something of a prequel, showing the two main characters on their first assignments. It contains animated violence and mayhem, some profanity and a small amount of nudity. The DVD release is rated 15 and up. The Pair consists of two 17-year-old junior level 'Trouble Consultants', a cross between a trouble-shooter and a secret agent. They work for an organization called World Welfare & Works Association or the 3WA which handles problems galaxy-wide. Their codename is 'Lovely Angels' but it's easy to see why they are eventually known as the Dirty Pair, not in a naughty way, but because they tend to leave ruination wherever they go. Yuri is the purple-haired and perky team member, she is occasionally zany and a bit distracted. Yuri tends to rely on her feminine wiles to get out of trouble, rarely plans ahead and is obsessed wit",
      "rating": 6,
      "runtime": 30,
      "status": "ended",
      "title": "Dirty Pair Flash",
      "trailer": null,
      "updated_at": "2015-11-12T10:30:48.000Z",
      "votes": 5,
      "year": 1994,
    },
    "type": "show",
  },
  Object {
    "score": 0.2593188,
    "show": Object {
      "aired_episodes": 62,
      "airs": Object {
        "day": "Sunday",
        "time": "00:15",
        "timezone": "America/New_York",
      },
      "available_translations": Array [
        "de",
        "en",
        "es",
        "ru",
      ],
      "certification": "TV-MA",
      "country": "us",
      "first_aired": "2006-08-05T22:00:00.000Z",
      "genres": Array [
        "animation",
        "comedy",
        "music",
      ],
      "homepage": "http://www.adultswim.com/shows/metal/",
      "ids": Object {
        "imdb": "tt0839188",
        "slug": "metalocalypse",
        "tmdb": 653,
        "trakt": 651,
        "tvdb": 79563,
        "tvrage": null,
      },
      "language": "en",
      "network": "Adult Swim",
      "overview": "Metalocalypse is an eleven-minute flash animated comedy airing on Cartoon Network's Adult Swim, created by Brendon Small and Tommy Blacha. The series follows the exploits of a fictitious half American/half Scandinavian death metal band known as Dethklok. The band and its members are huge the world over, but incredibly stupid - a death metal version of Spinal Tap. Watch as they leave a trail of destruction and mayhem in their wake with their ear-melting onslaught of metal!",
      "rating": 8.22737,
      "runtime": 11,
      "status": "ended",
      "title": "Metalocalypse",
      "trailer": null,
      "updated_at": "2017-04-18T09:29:27.000Z",
      "votes": 453,
      "year": 2006,
    },
    "type": "show",
  },
]
`;
